File: proof_unsat.py
#!/usr/bin/env python 

from z3 import *

set_option(max_args=10000000,
           max_lines=1000000,
           max_depth=10000000,
           max_visited=1000000,
           proof=True)
set_pp_option("max_indent", 0)
set_pp_option("max_width", 100000)

# s = Solver();
s = SolverFor("QF_LIA")
s.set("smt.arith.solver", 4)

# s.from_file("./tests/a.smt2")
# s.from_file("./tests/paper_example.smt2")
s.from_file("./tests/diff_int_logic.smt2")
if(s.check() == unsat):
    print(s.proof())

File: proof_unsat.cpp
#include <iostream>
#include <z3++.h>

int main(int argc, char ** argv){
  z3::set_param("proof", true);
  z3::set_param("pp.single_line", true);
  
  z3::context ctx;

  z3::solver s(ctx, "QF_UFLIA");
  s.from_file("./smt2_files/diff_int_logic.smt2");
  // s.from_file("./smt2_files/QF_UFLIA/mathsat/EufLaArithmetic/hard/hard10.smt2");
  // s.from_file("./smt2_files/QF_UFLIA/mathsat/Hash/hash_uns_03_03.smt2");
  if(s.check() == z3::unsat)
    std::cout << s.proof() << std::endl;
  return 0;
}
